bugsnag-android-core/src/main/java/com/bugsnag/android/EventFilenameInfo.kt

```diff
@@ -0,0 +1,165 @@
+package com.bugsnag.android
+
+import java.io.File
+import java.util.Locale
+import java.util.UUID
+
+/**
+ * Represents important information about an event which is encoded/decoded from a filename.
```

Comment:
This would be a useful place to detail the various things we're now encoding and the rationale behind them.

Reply:
Added some additional docs here